drivers/power/bd7181x-power.c

```diff
@@ -294,26 +296,44 @@ static u32 rsense_current_factor = 1000; // (/ factor)
 static struct tn_power_values_st tn_power_values;
 
 static void twonav_init_type(void) {
+
+	/* We can differentiate between mmcs using kernel parameter mmc_name
+	// Kingston 32GB: TX2932
+	// Kingston 16GB: TB2916
+	// Kingston 32GB: TA2932
+	// Toshiba 16GB:  016G30
+	*/
+
 	if(strstr(hwtype, "trailplus") != NULL) {
 		tn_power_values = TN_POWER_TRAIL;
-		rsense_capacity_factor = 360; // TODO: not known yet
-		rsense_current_factor = 1000; // TODO: not known yet
+		rsense_capacity_factor = 248; // verify
```

Comment:
why this values are not contained inside the same device struct as others?

Reply:
For the time being the battery is the same for Trail and TrailPlus but the Rsense correction is different thus to not duplicate the structure i let it seperatelly. We can loot at it on another iteration

Reply:
On the next change it'll be convenient to group those values in the same struct, as they could be different on other devices ( cross for example )